batchspawner/api.py

```diff
@@ -0,0 +1,16 @@
+import json
+from tornado import web
+from jupyterhub.apihandlers import  APIHandler, default_handlers
+
+class BatchSpawnerAPIHandler(APIHandler):
+    @web.authenticated
+    def post(self):
+        """POST set user's spawner port number"""
+        user = self.get_current_user()
+        data = self.get_json_body()
+        port = int(data.get('port', 0))
+        user.spawner.current_port = port
```

Comment:
When using `wrapspawner`, this fails because `user.spawner.current_port` needs to be proxied to `user.spawner.child_spawner.current_port`.  I made a quick fix to `wrapspawner` to proxy it there, but we can ask what's the best place to do this?

When I made `wrapspawner` proxy all attributes with __getattr__, it fails some other way which I haven't understood yet.

The main question is: should wrapspawner or batchspawner be responsible for this?  I would think wrapspawner, and how do we then avoid having to special case everything needed?  I'll return to this later.